### PR TITLE
Add tips for GC

### PR DESCRIPTION
**Pull request by SueChaplain**
- don't attempt to predict behaviour by using calls to `System.gc()` or finalizers
- If you're getting an OOM but you think your heap size is fine, check your Java dump, which might indicate something your code is doingthat's causing the problem.

Signed-off-by: SueChaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by dmitripivkine** [comment]
@amicic FYI

**Review by dmitripivkine — APPROVED** [review]

File: docs/allocation.md [hunk 1]
```diff
@@ -115,6 +115,8 @@ There are a number of advantages to using arraylets.
...
 
+When the Java heap runs out of space, `OutOfMemoryError` exceptions are generated. If you are confident that your heap settings are appropriate for your application but are still receiving an `OutOfMemoryError` exception, check the Java dump file that gets automatically generated when the error occurs. A Java dump file can tell you more about what your application was attempting to do at the time of the error. For example, see the [Java OutofMemoryError](dump_javadump.md#java-outofmemoryerror) scenario.
+
```
Comment (amicic) [inline docs/allocation.md:119]: in 'see the [Java OutofMemoryError]', second 'O' should be upper case

**Review by amicic — APPROVED** [review]

File: docs/allocation.md (hunk 1 above)
Comment (SueChaplain) [reply docs/allocation.md:119]: Thanks @amicic and @dmitripivkine for reviews/comments. Typo corrected.

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Pull request merged**